pkg/providers/v1/aws.go

```diff
@@ -4209,9 +4212,20 @@ func (c *Cloud) GetLoadBalancer(ctx context.Context, clusterName string, service
 	return status, true, nil
 }
 
+var inValidLBName = regexp.MustCompile(`[^a-zA-Z0-9-]`)
+
 // GetLoadBalancerName is an implementation of LoadBalancer.GetLoadBalancerName
 func (c *Cloud) GetLoadBalancerName(ctx context.Context, clusterName string, service *v1.Service) string {
-	// TODO: replace DefaultLoadBalancerName to generate more meaningful loadbalancer names.
+	if name, ok := service.Annotations[ServiceAnnotationLoadBalancerName]; ok {
+		if len(name) > 32 {
+			name = name[:32]
+			klog.Warningf("Load balancer name is over 32 characters, truncating: %s\n", name)
+		}
+		sanitised := inValidLBName.ReplaceAllString(name, "")
```

Comment:
test? Or comment what the regexp is doing. My reading of it is that it is removing all non-alphanumeric chars, is that right

Also, not sure if we should sanitize or error. Is sanitize what lb controller does?

Reply:
The LB controller sanitizes.  I considered erroring as well.  I am also not sure what would be preferrable... if there is ever a case where we want to look for a service given a load balancer, then erroring would be preferrable so that the elb name matches the annotation and the matching service can be easily found.  I'm not sure if that's important to support though.

Reply:
I dont like it but if lb controller is already doing it then best to be consistent. Still keep in mind accepting arbitrary inputs now means it will be very hard to change later.

Reply:
https://github.com/kubernetes-sigs/aws-load-balancer-controller/pull/1880 hasn't been included in an LB controller release yet, so its behavior could still be changed to error in this case

Reply:
Yeah good point, I'm leaning toward starting with error since it seems to be easier to switch to using a sanitized name later vs the other way around.

Reply:
As long as the error shows up somewhere discoverable (like as an event on the Service/Ingress object?) then I prefer error as well. Otherwise you could argue it's better to just sanitize and let the user figure out what happened. I don't fee lthat strongly either way, as long as cloudprovider and controller are consistent .